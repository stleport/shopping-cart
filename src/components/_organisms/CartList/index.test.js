import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Cart } from '.';

function CartListSetup() {
  const props = {
    cart: [
      {
        title: 'Item',
        productId: 1,
        quantity: 1,
        images: ['http://www.example.com'],
        description: 'Description'
      },
    ],
    addToCart: jest.fn(),
    addQuantity: jest.fn(),
    subQuantity: jest.fn(),
    removeFromCart: jest.fn(),
    total: 0
  }; 

  const CartListJSX = (
    <Cart {...props} />
  );

  const CartListComponent = shallow(CartListJSX);
  return {
    CartListComponent,
    CartListJSX,
    props
  };
}

describe('CartList', () => {
  it('renders correctly CartList component', () => {
    const { CartListJSX } = CartListSetup();
    const CartListComponent = renderer.create(CartListJSX).toJSON();
    expect(CartListComponent).toMatchSnapshot(); 
  });
});
