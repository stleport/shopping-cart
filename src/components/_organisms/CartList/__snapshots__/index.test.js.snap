// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartList renders correctly CartList component 1`] = `
Array [
  <h1>
    Panier
  </h1>,
  <div
    className="st-Items"
  >
    <div
      className="st-Item"
    >
      <div
        className="st-Item__image--small"
      >
        <img
          alt="Item"
          className="image"
          onError={[Function]}
          src="http://www.example.com"
        />
      </div>
      <div
        className="st-Item__content"
      >
        <h2>
          Item
        </h2>
        <p
          className="description"
        >
          Description
        </p>
        <div
          className="st-Item__pricing"
        >
          Quantité : 
          1
        </div>
        <div
          className="st-Item__extra"
        >
          <button
            className="st-Edit-button__button st-Edit-button__button--dark buttontrue"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="st-Edit-button__icon--white"
              fill="white"
            >
              minus.svg
            </svg>
          </button>
          <button
            className="st-Edit-button__button st-Edit-button__button--light buttontrue"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="st-Edit-button__icon--dark"
              fill="dark"
            >
              plus.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <h3>
    Total : 
    0.00 €
  </h3>,
]
`;
